### PR TITLE
Feature/database schema

### PR DESCRIPTION
**Pull request by oklymchuk**

### CODE REVIEW

**Comment by radikh** [comment]
You have conflicts with master now

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        - "gpsnet"
+      env_file:
+        - ".env"
+      volumes:
+        - ./db-data:/var/lib/postgresql/data
+        - ./migrator/init:/docker-entrypoint-initdb.d/
+      ports:
+        - 5432:5432
+
+    migrator:
```
Comment (dmitriyminer) [inline docker-compose.yml]: What do you think about using a command for migrations?
https://github.com/lvl484/positioning-filter/pull/7/files#diff-04c6e90faac2675aa89e2176d2eec7d8R70

File: migrator/migrations/3_add_grant_user_to_users.down.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+REVOKE SELECT, INSERT, UPDATE  ON users FROM gateway;
```
Comment (dmitriyminer) [inline migrator/migrations/3_add_grant_user_to_users.down.sql]: What about DELETE?

File: config/config.go [hunk 3]
```diff
@@ -1,4 +0,0 @@
-// Package config is responsible for loading user-manager application config.
-// Basic configuration like consul credentials and address, http port to listen for requests,
-// postgres schema name, credentials, and client timeout are read from environment variables.
```
Comment (dmitriyminer) [inline config/config.go]: Code from the previous contributor removed
Why?

File: migrator/init/user_for_api.sh [hunk 4]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+psql -c "CREATE USER gateway WITH PASSWORD $UM_PASSWORD;"
```
Comment (dmitriyminer) [inline migrator/init/user_for_api.sh]: WDYT about renaming user 'gateway' to 'um_user'?
Pattern: [app abbreviation] + '_user

File: config/config.go (hunk 3 above)
Comment (oklymchuk) [reply config/config.go]: Because this code was merge by mistake to this branch

File: migrator/init/user_for_api.sh (hunk 4 above)
Comment (oklymchuk) [reply migrator/init/user_for_api.sh]: Agree, fix it

File: migrator/migrations/3_add_grant_user_to_users.down.sql (hunk 2 above)
Comment (oklymchuk) [reply migrator/migrations/3_add_grant_user_to_users.down.sql]: By the grant we didn't give this privilege to this user

File: docker-compose.yml (hunk 1 above)
Comment (oklymchuk) [reply docker-compose.yml]: It  will not allow to us make dockers automation containers up for us in future with others services

File: config/config.go (hunk 3 above)
Comment (dmitriyminer) [reply config/config.go]: As soon as you merge PR, 93 lines of code from @i3odja will be deleted
Is it okay?

Comment (oklymchuk) [reply config/config.go]: fix it

File: .gitignore [hunk 5]
```diff
@@ -1,3 +1,17 @@
+<<<<<<< HEAD
```
Comment (dmitriyminer) [inline .gitignore]: Unresolved conflicts

File: migrator/migrations/1_create_users_table.up.sql [hunk 6]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE public.users
+(
+    id uuid NOT NULL,
+    user_name text NOT NULL,
```
Comment (dmitriyminer) [inline migrator/migrations/1_create_users_table.up.sql]: Text used for long strings VARCHAR is much better

File: config/config.go (hunk 3 above)
Comment (dmitriyminer) [reply config/config.go]: `go.mod` marked as removed

File: migrator/init/user_for_api.sh [hunk 7]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+psql -c "CREATE USER um_user WITH PASSWORD $UM_PASSWORD;"
```
Comment (crnopster) [inline migrator/init/user_for_api.sh]: Why not to make um_user env variable just like $UM_PASSWORD?

File: migrator/migrations/1_create_users_table.up.sql [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    "password" text NOT NULL,
+    email text NOT NULL,
+    first_name text NOT NULL,
+    last_name text NOT NULL,
+    phone text NOT NULL,
```
Comment (crnopster) [inline migrator/migrations/1_create_users_table.up.sql]: TEXT is not good for this rows. Better use VARCHAR

File: migrator/migrations/3_add_grant_user_to_users.down.sql (hunk 2 above)
Comment (radikh) [reply migrator/migrations/3_add_grant_user_to_users.down.sql]: Why don't we? Delete operation will be required for REST interface and command line tool.

Comment (oklymchuk) [reply migrator/migrations/3_add_grant_user_to_users.down.sql]: fix it

File: config/config.go (hunk 3 above)
Comment (oklymchuk) [reply config/config.go]: fix it

File: .gitignore (hunk 5 above)
Comment (oklymchuk) [reply .gitignore]: fix it

File: migrator/migrations/1_create_users_table.up.sql (hunk 6 above)
Comment (dmitriyminer) [reply migrator/migrations/1_create_users_table.up.sql]: What about `username`, `first_name`, `last_name`, etc?
For example: 
```
Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer lobortis aliquam elit, ut molestie mi. Sed semper orci ut arcu accumsan lacinia. Vivamus mollis tellus molestie nulla iaculis luctus. Morbi at suscipit risus. Ut aliquam ultrices commodo. Quisque hendrerit ornare luctus. Fusce scelerisque lectus ac tellus finibus sagittis. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia Curae; Sed tincidunt lacus quis lectus luctus auctor.
```

Comment (oklymchuk) [reply migrator/migrations/1_create_users_table.up.sql]: The longest name used by anyone is Adolph Blaine Charles David Earl Frederick Gerald Hubert Irvin John Kenneth Lloyd Martin Nero Oliver Paul Quincy Randolph Sherman Thomas Uncas Victor William Xerxes Yancy Wolfeschlegelsteinhausenbergerdorff, Senior, who was born at Bergedorf, near Hamburg, Germany, on 29 Feb. 1904. On printed forms he uses only his eighth and second Christian names and the first 35 letters of his surname. The full version of the name of 590 letters appeared in the 12th edition of The Guinness Book of Records. He now lives in Philadelphia, Pennsylvania, U.S.A., and has shortened his surname to Mr. Wolfe + 585, Senior.  http://hoaxes.org/weblog/comments/worlds_longest_surname

Comment (oklymchuk) [reply migrator/migrations/1_create_users_table.up.sql]: Do you think i should change to varchar?

Comment (oklymchuk) [reply migrator/migrations/1_create_users_table.up.sql]: fix it

File: migrator/migrations/2_add_constraint_on_users.up.sql [hunk 9]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE public.users ADD CONSTRAINT users_un_unigue UNIQUE (user_name);
+ALTER TABLE public.users ADD CONSTRAINT users_email_unigue UNIQUE (email);
+ALTER TABLE public.users ADD CONSTRAINT users_phone_unigue UNIQUE (phone);
```
Comment (radikh) [inline migrator/migrations/2_add_constraint_on_users.up.sql]: As I remember you can add UNIQUE when create the table. So you don't really need these constraints to be in separate migrations.

Comment (oklymchuk) [reply migrator/migrations/2_add_constraint_on_users.up.sql]: Fixed

File: migrator/migrations/1_create_users_table.up.sql (hunk 6 above)
Comment (radikh) [reply migrator/migrations/1_create_users_table.up.sql]: > The longest name used by anyone is Adolph Blaine Charles David Earl Frederick Gerald Hubert Irvin John Kenneth Lloyd Martin Nero Oliver Paul Quincy Randolph Sherman Thomas Uncas Victor William Xerxes Yancy Wolfeschlegelsteinhausenbergerdorff, Senior, who was born at Bergedorf, near Hamburg, Germany, on 29 Feb. 1904. On printed forms he uses only his eighth and second Christian names and the first 35 letters of his surname. The full version of the name of 590 letters appeared in the 12th edition of The Guinness Book of Records. He now lives in Philadelphia, Pennsylvania, U.S.A., and has shortened his surname to Mr. Wolfe + 585, Senior. http://hoaxes.org/weblog/comments/worlds_longest_surname

Good point. And misunderstanding. Username is a synonym of login in this scope, and it shouldn't be long. For stuff you mentioned user can use First name or lat name fields.

File: migrator/migrations/1_create_users_table.up.sql [hunk 10]
```diff
@@ -0,0 +1,13 @@
+CREATE TABLE public.users
+(
+    id uuid NOT NULL,
+    user_name text NOT NULL,
+    "password" text NOT NULL,
+    email text NOT NULL,
```
Comment (radikh) [inline migrator/migrations/1_create_users_table.up.sql]: These two also should be varchar.

Comment (oklymchuk) [reply migrator/migrations/1_create_users_table.up.sql]: It is fixed

File: migrator/migrations/1_create_users_table.up.sql [hunk 11]
```diff
@@ -0,0 +1,16 @@
+CREATE TABLE public.users
```
Comment (dmitriyminer) [inline migrator/migrations/1_create_users_table.up.sql]: `CREATE TABLE` use the `public` schema but  `DROP TABLE` does not

Comment (crnopster) [reply migrator/migrations/1_create_users_table.up.sql]: by default CREATE TABLE uses public schema

File: migrator/migrations/1_create_users_table.up.sql [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+(
+    id uuid NOT NULL,
+    user_name varchar(20) NOT NULL,
+    "password" varchar(20) NOT NULL,
```
[the remark this answers is not in the document]
Comment (oklymchuk) [reply migrator/migrations/1_create_users_table.up.sql]: password?

File: migrator/migrations/1_create_users_table.up.sql [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+    email varchar(70) NOT NULL,
+    first_name varchar(20) NOT NULL,
+    last_name varchar(50) NOT NULL,
+    phone varchar(12) NOT NULL,
```
[the remark this answers is not in the document]
Comment (oklymchuk) [reply migrator/migrations/1_create_users_table.up.sql]: fixed

File: migrator/migrations/1_create_users_table.up.sql (hunk 12 above)
[the remark this answers is not in the document]
Comment (oklymchuk) [reply migrator/migrations/1_create_users_table.up.sql]: fixed

File: migrator/migrations/1_create_users_table.up.sql (hunk 11 above)
Comment (oklymchuk) [reply migrator/migrations/1_create_users_table.up.sql]: fixed

**Pull request merged**